internal/controllers/machinedeployment/machinedeployment_rollout_planner_test.go

```diff
@@ -499,10 +501,71 @@ func machineSetControllerMutator(log *fileLogger, ms *clusterv1.MachineSet, scop
 	// Note: this should not be implemented in production code
 	ms.Status.Replicas = ptr.To(int32(len(scope.machineSetMachines[ms.Name])))
 
+	// TODO(in-place): when implementing in-place changes in the MachineController code make sure to:
+	//  - detect if there are replicas still pending AcknowledgeMove first, including also handling cleanup of the pendingAcknowledgeMoveAnnotationName on machines
+	//  - when deleting or moving
+	//  	- first move if possible, then delete
+	// 		- move machines should be capped to avoid unnecessary in-place upgrades (in case of scale down in the middle of rollouts); remaining part should be deleted
+	//      - do not move machines pending a move Acknowledge, with an in-place upgrade in progress, deleted or marked for deletion, unhealthy
+
 	// Sort machines to ensure stable results of move/delete operations during tests.
-	// Note: this should not be implemented in production code
+	// TODO(in-place): this should not be implemented in production code for the MachineSet controller
```

Comment:
Why did this "should not" become a TODO?

Reply:
It is just a reminder that in-place code does not need sorting the same wait it is implemented in tests.

Reply:
All good, thx!